core/src/test/java/cucumber/runtime/table/TableDifferTest.java

```diff
@@ -229,6 +230,41 @@ public void should_diff_with_empty_list() {
     }
 
     @Test(expected = TableDiffException.class)
+    public void should_diff_with_empty_table() {
+        try {
+            DataTable emptyTable = DataTable.create(Collections.emptyList());
+            table().diff(emptyTable);
+        } catch (IndexOutOfBoundsException e) {
+            String expected = "" +
+                "Tables were not identical:\n" +
+                "    - | Aslak | aslak@email.com | 123 |\n" +
+                "    - | Joe   | joe@email.com   | 234 |\n" +
+                "    - | Bryan | bryan@email.org | 456 |\n" +
+                "    - | Ni    | ni@email.com    | 654 |\n";
+            assertEquals(expected, e.getMessage());
+            throw e;
+        }
+    }
+
+    @Test
+    public void empty_list_should_not_diff_with_empty_table() {
+        try {
+            List<List<String>> emptyList = new ArrayList<List<String>>();
+            DataTable emptyTable = DataTable.create(Collections.emptyList());
+            emptyTable.diff(emptyList);
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
In this test case we want to compare the empty table another another empty table. This should not result in any exception.